File: src/main/java/de/bonndan/nivio/output/map/NText.java
package de.bonndan.nivio.output.map;

import j2html.tags.ContainerTag;

class NText extends Component {

    private final ItemMapItem item;
    final long x, y;
    private String className;
    private final int width;

    NText(ItemMapItem item, long x, long y, String className, int width) {
        this.item = item;
        this.x = x;
        this.y = y;
        this.className = className;
        this.width = width;
    }

    public ContainerTag render() {

        return SvgTagCreator.text(item.name)
                .attr("x", x)
                .attr("y", y==0 ? "0.3em" : y)
                .attr("width", width)
                .attr("class", className)
                .attr("text-anchor", "middle");
    }
}

File: src/main/app/src/hexmap/NText.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NText extends Component {
  static propTypes = {
    item: PropTypes.object,
    x: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    y: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    className: PropTypes.string,
    width: PropTypes.number,
    onClick: PropTypes.func,
  };

  render() {
    const { item, x, y, className, width, onClick } = this.props;
    return (
      <text
        x={x || 0}
        y={y ? y : '0.3em'}
        width={width}
        className={className}
        textAnchor='middle'
        onClick={onClick}
      >
        {item.name}
      </text>
    );
  }
}

export default NText;
